Arms APL Update

Awesome2damax opened the pull request:
Update to the Arms APL. Very large DPS increases with Rend and Massacre, and a smaller DPS increase to Fervor of Battle. Changes to better account for a few different Major Essence choices, leading to a DPS increase for Condensed Life Force and Blood of the Enemy. Updated a few racial lines for the better. Most of these changes are larger increases the more Expedient you have.

Mostly comprehensive change list: 
1) Rewrote precombat lines for Memory of Lucid Dreams and Guardian of Azeroth to account for shorter fight timings, leading to large dps increases for 2 minute and 3 minute fights. 
2) Added conditionals to force certain potions for precombat based on essence choice.
3) Updated the second potion line.
4) Updated the usage of Ashvane's Razor Coral.
5) Updated Blood of the Enemy, Guardian of Azeroth, and Memory of Lucid Dreams lines.
6) Added Rend to the Execute section of the APL as well as changed its priority in the Single Target section.
7) Changed some conditionals for casting Execute.
8) Changed some conditionals for casting Whirlwind.

Pastebin version of the new APL : https://pastebin.com/b0sn1668

5 minute APL comparison sim with super high end gear Lucid - https://www.raidbots.com/simbot/report/eCUidknezTcLq8vc4GETpk
5 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/tgPA3GebckYJ3nSDeZEL8U
3 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/aYxc3zuTvnyKhtmgPTL4x9
2 minute APL comparison sim with T25 Arms Mythic default CLF - https://www.raidbots.com/simbot/report/79bdFNoXx3wfBAmkTFyyy

Awesome2damax committed:
Arms APL Update

Update to the Arms APL. Very large DPS increases with Rend and Massacre, and a smaller DPS increase to Fervor of Battle. Changes to better account for a few different Major Essence choices, leading to a DPS increase for Condensed Life Force and Blood of the Enemy. Updated a few racial lines for the better. Most of these changes are larger increases the more Expedient you have.

Mostly comprehensive change list: 
1) Rewrote precombat lines for Memory of Lucid Dreams and Guardian of Azeroth to account for shorter fight timings, leading to large dps increases for 2 minute and 3 minute fights. 
2) Added conditionals to force certain potions for precombat based on essence choice.
3) Updated the second potion line.
4) Updated the usage of Ashvane's Razor Coral.
5) Updated Blood of the Enemy, Guardian of Azeroth, and Memory of Lucid Dreams lines.
6) Added Rend to the Execute section of the APL as well as changed its priority in the Single Target section.
7) Changed some conditionals for casting Execute.
8) Changed some conditionals for casting Whirlwind.

5 minute APL comparison sim with super high end gear Lucid - https://www.raidbots.com/simbot/report/eCUidknezTcLq8vc4GETpk
5 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/tgPA3GebckYJ3nSDeZEL8U
3 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/aYxc3zuTvnyKhtmgPTL4x9
2 minute APL comparison sim with T25 Arms Mythic default CLF - https://www.raidbots.com/simbot/report/79bdFNoXx3wfBAmkTFyyy

Awesome2damax committed:
Arms APL Update (Fixed missing commas)

Update to the Arms APL. Very large DPS increases with Rend and Massacre, and a smaller DPS increase to Fervor of Battle. Changes to better account for a few different Major Essence choices, leading to a DPS increase for Condensed Life Force and Blood of the Enemy. Updated a few racial lines for the better. Most of these changes are larger increases the more Expedient you have.

Mostly comprehensive change list:

Rewrote precombat lines for Memory of Lucid Dreams and Guardian of Azeroth to account for shorter fight timings, leading to large dps increases for 2 minute and 3 minute fights.
Added conditionals to force certain potions for precombat based on essence choice.
Updated the second potion line.
Updated the usage of Ashvane's Razor Coral.
Updated Blood of the Enemy, Guardian of Azeroth, and Memory of Lucid Dreams lines.
Added Rend to the Execute section of the APL as well as changed its priority in the Single Target section.
Changed some conditionals for casting Execute.
Changed some conditionals for casting Whirlwind.
5 minute APL comparison sim with super high end gear Lucid - https://www.raidbots.com/simbot/report/eCUidknezTcLq8vc4GETpk
5 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/tgPA3GebckYJ3nSDeZEL8U
3 minute APL comparison sim with T25 Arms Mythic default Lucid - https://www.raidbots.com/simbot/report/aYxc3zuTvnyKhtmgPTL4x9
2 minute APL comparison sim with T25 Arms Mythic default CLF - https://www.raidbots.com/simbot/report/79bdFNoXx3wfBAmkTFyyy

Awesome2damax committed:
Arms APL Update attempt 3

I'm bad at this and maybe its fixed

Awesome2damax committed:
Update Arms APL Attempt 4

Trying to fix the errors again, APL itself should be sound

Awesome2damax committed:
Arms APL Update Added missing bracket

Added a missing ) on the precombat guardian of azeroth line

Melekus commented on engine/class_modules/sc_warrior.cpp:
```diff
@@ -5078,16 +5078,31 @@ void warrior_t::default_apl_dps_precombat()
+    precombat->add_action( this, "potion", "if=!essence.memory_of_lucid_dreams.major&!essence.condensed_lifeforce.major" );
   }
 
-  precombat->add_action( "potion" );
```
This will remove the usage of potion in precombat for all specs, not just Fury and Arms.

Awesome2damax replied:
The newest commit in this pull request should fix the stated issue, as well as fix issues with the precombat for both Fury and Arms that I overlooked the first time.

Awesome2damax committed:
Arms APL Update

Fixed Precombat lines

Awesome2damax committed:
Update Arms APL

I used add_action instead of add_talent for Rend, now fixed.

Awesome2damax committed:
Arms APL Update

Added a Time to Target Percent conditional to Colossus Smash in the Single Target section, and added a better Time to Die conditional to the second potion usage. Not very relevant on a 5 minute sim but leads to good increases on shorter sims.

Pastebin with compiled sim comparisons for Time to Target Percent = https://pastebin.com/GXceD7VD
